src/GitArtifactGitRepository.php

```diff
@@ -0,0 +1,348 @@
+    $argsCommand = ['log', $options];
+    if (!empty($revisionRange)) {
+      $argsCommand = array_merge($argsCommand, [$revisionRange]);
+    }
+    $argsCommand = array_merge($argsCommand, ['--end-of-options'], $paths);
+    $output = $this->extractFromCommand($argsCommand);
+
+    if (is_null($output)) {
+      return [];
+    }
+
+    return $output;
+  }
+
+  /**
+   * Reset command.
+   *
+   * @param array<mixed> $options
+   *   Options.
+   *
+   * @return $this
+   *   Git repo.
+   *
+   * @throws \CzProject\GitPhp\GitException
+   */
+  public function reset(array $options = []): GitArtifactGitRepository {
+    $this->run('reset', $options);
+
+    return $this;
+  }
+
+  /**
+   * Clean command.
+   *
+   * @param array<mixed> $options
+   *   Options.
+   *
+   * @return $this
+   *   Git repo.
+   *
+   * @throws \CzProject\GitPhp\GitException
+   */
+  public function clean(array $options = []): GitArtifactGitRepository {
+    $this->run('clean', $options);
+
+    return $this;
+  }
+
+  /**
+   * Switch to new branch.
+   *
+   * @param string $branchName
+   *   Branch name.
+   * @param bool $createNew
+   *   Optional flag to also create a branch before switching. Default false.
+   *
+   * @return GitArtifactGitRepository
+   *   The git repository.
+   *
+   * @throws \CzProject\GitPhp\GitException
+   */
+  public function switchToBranch(string $branchName, bool $createNew = FALSE): GitArtifactGitRepository {
+    if (!$createNew) {
+      return $this->checkout($branchName);
+    }
+
+    return $this->createBranch($branchName, TRUE);
+  }
+
+  /**
+   * Remove branch.
+   *
+   * @param string $name
+   *   Branch name.
+   * @param bool $force
+   *   Force remove or not.
+   *
+   * @return GitArtifactGitRepository
+   *   Git repository
+   *
+   * @throws \CzProject\GitPhp\GitException
+   */
+  public function removeBranch($name, bool $force = FALSE): GitArtifactGitRepository {
+    if (empty($name)) {
+      return $this;
+    }
+
+    if (!$force) {
+      return parent::removeBranch($name);
+    }
+
+    $this->run('branch', ['-D' => $name]);
+
+    return $this;
+  }
+
+  /**
+   * Commit all files to git repo.
+   *
+   * @param string $message
+   *   The commit message.
+   *
+   * @return array<string>
+   *   The changes.
+   *
+   * @throws \CzProject\GitPhp\GitException
+   */
+  public function commitAllChanges(string $message): array {
+    $this->addAllChanges();
+
+    // We do not use commit method because we need return the output.
+    return $this->execute('commit', '--allow-empty', [
+      '-m' => $message,
+    ]);
+  }
+
+  /**
+   * Git ls-files command.
+   *
+   * @param array<mixed> $options
+   *   Options.
+   * @param array<string> $files
+   *   Files to show.
+   *
+   * @return string[]|null
+   *   Files.
+   *
+   * @throws \CzProject\GitPhp\GitException
+   */
+  public function lsFiles(array $options = [], array $files = []): ?array {
```

Comment:
please name the method names as what they are: `listFiles()`. Also - what is `options`? We should not ever pass any CLI git options to these methods - these methods should abstract all of that!